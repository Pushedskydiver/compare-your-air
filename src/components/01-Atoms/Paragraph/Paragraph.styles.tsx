import styled, { css } from 'styled-components';
import { typography } from '../../../styles/typography.styles';
import spacing from '../../../styles/spacing.styles';

interface ParagraphProps {
  noMargin?: boolean,
  small?: boolean,
  medium?: boolean,
  bold?: boolean,
  uppercase?: boolean,
  align?: string,
}

const Paragraph = styled.p<ParagraphProps>`
  font-family: var(--font);
  line-height: ${typography.lineHeight.body};

  &:last-of-type {
    margin-bottom: 0;
  }

  ${props => !props.noMargin && css`
    margin: 0 0 ${(spacing.margin * 1.5)}px 0;
  `}

  ${props => props.small && css`
    font-size: ${typography.body.size.small};
  `}

  ${props => props.medium && css`
    font-size: ${typography.body.size.medium};
  `}

  ${props => props.bold && css`
    font-weight: ${typography.weight.medium};
  `}

  ${props => props.uppercase && css`
    text-transform: uppercase;
  `}

  ${props => props.align && css`
    text-align: ${props.align};
  `}
`;

export default Paragraph;
