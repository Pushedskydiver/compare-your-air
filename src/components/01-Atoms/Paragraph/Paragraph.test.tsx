import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Paragraph from './Paragraph.styles';

afterEach(() => {
  cleanup();
});

describe('<Paragraph />', () => {
  it('should match default paragraph snapshot', () => {
    const { container } = render(
      <Paragraph>This is a p tag</Paragraph>
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot of paragraph with no margin', () => {
    const { container } = render(
      <Paragraph noMargin>This is a p tag</Paragraph>
    );

    expect(container).toMatchSnapshot();
  });
})