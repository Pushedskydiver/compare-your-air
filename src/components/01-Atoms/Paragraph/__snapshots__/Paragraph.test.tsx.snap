// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should match default paragraph snapshot 1`] = `
<div>
  <p
    class="sc-AxjAm suCXS"
  >
    This is a p tag
  </p>
</div>
`;

exports[`<Paragraph /> should match snapshot of paragraph with no margin 1`] = `
<div>
  <p
    class="sc-AxjAm lelnYw"
  >
    This is a p tag
  </p>
</div>
`;
